CUDADataFormats/HGCal/interface/HGCCLUEGPUProduct.h

```diff
@@ -6,16 +6,15 @@
 #include "HeterogeneousCore/CUDAUtilities/interface/device_unique_ptr.h"
 
 #include "CUDADataFormats/HGCal/interface/HGCCLUESoA.h"
-#include "CUDADataFormats/HGCal/interface/ConstHGCCLUESoA.h"
 #include "CUDADataFormats/HGCal/interface/HGCUncalibRecHitSoA.h"
 
 class HGCCLUEGPUProduct {
 public:
   HGCCLUEGPUProduct() = default;
   explicit HGCCLUEGPUProduct(uint32_t nhits, const cudaStream_t &stream) : nhits_(nhits) {
-    size_tot_ = std::accumulate(sizes_.begin(), sizes_.end(), 0);
-    pad_ = ((nhits - 1) / 32 + 1) * 32; //align to warp boundary (assumption: warpSize = 32)
-    mMemCLUEDev = cms::cuda::make_device_unique<std::byte[]>(pad_ * size_tot_, stream);
+    /* CUDA allocations are already aligned */
+    mMemCLUEDev = cms::cuda::make_device_unique<std::byte[]>(
```

Comment:
I think we mean different things here (I have to admit my comment in the code was misleading). I was not trying to align the overall memory block. I was instead making sure the total size of the allocated memory is a multiple of the *warpSize* (32), and further making sure that each variable within the SoA is also aligned, so that no warp must allocate unnecessary memory blocks (that is why I am later using ```pad_``` when defining the layout of the SoA). If I would not do this, the only variable of the SoA which would surely be aligned would be the first.